rust/agama-server/src/agama-web-server.rs

```diff
@@ -100,30 +98,53 @@ struct ServeArgs {
 }
 
 impl ServeArgs {
-    /// Builds an SSL acceptor using a provided SSL certificate or generates a self-signed one
-    fn ssl_acceptor(&self) -> Result<SslAcceptor, openssl::error::ErrorStack> {
-        let mut tls_builder = SslAcceptor::mozilla_modern_v5(SslMethod::tls_server())?;
+    /// Returns true of given path to certificate points to an existing file
+    fn valid_cert_path(&self) -> bool {
+        self.cert.clone().is_some_and(|c| Path::new(&c).exists())
+    }
 
-        if let (Some(cert), Some(key)) = (self.cert.clone(), self.key.clone()) {
-            tracing::info!("Loading PEM certificate: {}", cert);
-            tls_builder.set_certificate_file(PathBuf::from(cert), SslFiletype::PEM)?;
+    /// Returns true of given path to key points to an existing file
+    fn valid_key_path(&self) -> bool {
+        self.key.clone().is_some_and(|k| Path::new(&k).exists())
+    }
 
-            tracing::info!("Loading PEM key: {}", key);
-            tls_builder.set_private_key_file(PathBuf::from(key), SslFiletype::PEM)?;
-        } else {
-            let (cert, key) = agama_server::cert::create_certificate()?;
+    /// Takes options provided by user and loads / creates Certificate struct according to them
+    fn to_certificate(&self) -> anyhow::Result<agama_server::cert::Certificate> {
+        return if self.valid_cert_path() && self.valid_key_path() {
```

Comment:
This condition is actually readable :+1:

Reply:
happens to me sometimes ;-)